### PR TITLE
Add migration guide for post-3.19 deprecation cleanup

### PR DESCRIPTION
**Pull request by Piinks**
Fixes https://github.com/flutter/flutter/issues/143956

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

File: src/release/breaking-changes/3-19-deprecations.md [hunk 1]
```diff
@@ -0,0 +1,728 @@
...
+
+---
+
+### `Scrollbar.showTrackOnHover`
+
+Package: flutter
+Supported by Flutter Fix: yes
+
+The `showTrackOnHover` property of `Scrollbar`, and its associated component theme,
+`ScrollbarThemeData.showTrackOnHover`, was replaced by the stateful property
```
Comment (sfshaza2) [inline src/release/breaking-changes/3-19-deprecations.md:385]:
```suggestion
`ScrollbarThemeData.showTrackOnHover`, were replaced by the stateful property
```

File: src/release/breaking-changes/3-19-deprecations.md [hunk 2]
```diff
@@ -0,0 +1,728 @@
...
+[#79310]: {{site.repo.flutter}}/pull/79310
+[#143983]: {{site.repo.flutter}}/pull/143983
+
+---
+
+The [previously announced][] deprecations for context menus, relating to `ToolbarOptions`
+as well as parts of `TextSelectionController` and `SelectableRegionState` were not
+removed this cycle, to allow more time for migration.
+Expect these deprecations to be removed in the next cycle, which will
+be announced again when it comes.
```
Comment (sfshaza2) [inline src/release/breaking-changes/3-19-deprecations.md:720]:
```suggestion
be announced again when the time comes.
```

**Review by sfshaza2 — APPROVED** [review]
This looks great, @Piinks! Just a couple little grammatical things.

**Review by parlough — APPROVED** [review]
Thanks!

**Pull request merged**